File: Assets/src/model/tasks/ChaseDynamicTargetTask.cs
using System;

[System.Serializable]
internal class ChaseDynamicTargetTask : ChaseTargetTask {
  private Func<Body> targetDecider;

  public ChaseDynamicTargetTask(Actor actor, Func<Body> targetDecider) : base(actor, targetDecider()) {
    this.targetDecider = targetDecider;
  }

  public override void PreStep() {
    this.targetBody = targetDecider();
    base.PreStep();
  }
}

File: web/src/model/tasks/ChaseDynamicTargetTask.ts
import { ChaseTargetTask } from './ChaseTargetTask';
import type { Actor } from '../Actor';
import type { Body } from '../Body';

export class ChaseDynamicTargetTask extends ChaseTargetTask {
  private targetDecider: () => Body;

  constructor(actor: Actor, targetDecider: () => Body) {
    super(actor, targetDecider());
    this.targetDecider = targetDecider;
  }

  preStep(): void {
    this.targetBody = this.targetDecider();
    super.preStep();
  }
}
